packages/xo-web/src/common/search-bar.js

```diff
@@ -0,0 +1,88 @@
+import _ from 'intl'
+import classNames from 'classnames'
+import DropdownMenu from 'react-bootstrap-4/lib/DropdownMenu' // https://phabricator.babeljs.io/T6662 so Dropdown.Menu won't work like https://react-bootstrap.github.io/components.html#btn-dropdowns-custom
+import DropdownToggle from 'react-bootstrap-4/lib/DropdownToggle' // https://phabricator.babeljs.io/T6662 so Dropdown.Toggle won't work https://react-bootstrap.github.io/components.html#btn-dropdowns-custom
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import { Dropdown, MenuItem } from 'react-bootstrap-4/lib'
+import { Input as DebouncedInput } from 'debounce-input-decorator'
+import { isEmpty, map } from 'lodash'
+
+import Button from './button'
+import Icon from './icon'
+import Tooltip from './tooltip'
+
+export default class SearchBar extends Component {
+  static propTypes = {
+    filters: PropTypes.object,
+    onChange: PropTypes.func.isRequired,
+    value: PropTypes.string.isRequired,
+  }
+
+  _cleanFilter = () => this._setFilter('')
+
+  _setFilter = filterValue => {
+    const filter = this.refs.filter.getWrappedInstance()
+    filter.value = filterValue
+    filter.focus()
+    this.props.onChange(filterValue)
+  }
+
+  _onChange = event => {
+    this.props.onChange(event.target.value)
+  }
+
+  focus() {
+    this.refs.filter.getWrappedInstance().focus()
+  }
+
+  render() {
+    const { props } = this
+
+    return (
+      <div className={classNames('input-group', props.className)}>
+        {isEmpty(props.filters) ? (
+          <span className='input-group-addon'>
+            <Icon icon='search' />
+          </span>
+        ) : (
+          <span className='input-group-btn'>
+            <Dropdown id='filter'>
+              <DropdownToggle bsStyle='info'>
+                <Icon icon='search' />
+              </DropdownToggle>
+              <DropdownMenu>
+                {map(props.filters, (filter, label) => (
+                  <MenuItem key={label} onClick={() => this._setFilter(filter)}>
```

Comment:
This should probably be a computed.

Reply:
I didn't implement this React component, i just mutualized to use it. The change will not be linked to the issue which i'm fixing. I can do it in an other PR if it's necessary.

Reply:
Makes sense, you can ignore it then :slightly_smiling_face: